.gitattributes

```diff
@@ -1,2 +1,4 @@
 *.png filter=lfs diff=lfs merge=lfs -text
 *.jpg filter=lfs diff=lfs merge=lfs -text
+
+api/** linguist-generated=true
```

Comment:
Mark all files in `api` as generated. This means GitHub will not show the diff by default.

Reply:
What files was github showing? It seems better to do this by ensuring all files have the magic comment `// Code generated`

Reply:
It was showing the shorter files; the larger ones were hidden already.

I don't think there's an option to control this based on the content of the file.

I've got no strong opinions here, happy to remove it again.

Some examples:

<img width="1072" alt="Screenshot 2025-03-28 at 10 19 54 AM" src="https://github.com/user-attachments/assets/c3539d27-b8dc-403a-a9b7-ae26a1b4d02e" />
<img width="1065" alt="Screenshot 2025-03-28 at 10 19 49 AM" src="https://github.com/user-attachments/assets/5b08d3be-2a20-4b49-aea9-d7657dc6c431" />

Reply:
Any Go files with that string in the first 40 lines should be detected as generated: https://github.com/github-linguist/linguist/blob/main/lib/linguist/generated.rb#L334
I believe protoc and gomock adds this to all their output but maybe some other tools don't?

Reply:
I've removed it now.